Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -22,7 +22,9 @@ YUI.add('ez-roleserversideviewservice', function (Y) {
     Y.eZ.RoleServerSideViewService = Y.Base.create('roleServerSideViewService', Y.eZ.ServerSideViewService, [], {
         initializer: function () {
             this.on('*:contentDiscover', function (e) {
-                e.config.contentDiscoveredHandler = Y.bind(this._assignRole, this);
+                if (e.config.contentDiscoveredHandler === undefined) {
```

Comment:
this kind of logic feels wrong. I mean this has a very high *WTF potential* if you come back in a month ;-)

joke aside, you should introduce a method called for instance `_isDiscoveringForAssigningRole` (feel free to suggest a better name :)). Also, instead of checking the `contentDiscoveredHandler`, I would rather check that `e.config.data` contains a `roleId`.